weak_supervision/data/dataset_readers/semantic_parsing/latent_alignment.py

```diff
@@ -8,34 +8,51 @@
 from allennlp.data.dataset_readers.dataset_reader import DatasetReader
 from allennlp.data.fields import TextField, Field, ListField, MetadataField
 from allennlp.data.instance import Instance
-from allennlp.data.tokenizers import Token, WordTokenizer
+from allennlp.data.tokenizers import Token, Tokenizer, WordTokenizer
 from allennlp.data.tokenizers.word_splitter import JustSpacesWordSplitter
-from allennlp.data.token_indexers import SingleIdTokenIndexer
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+def preprocess_tokens(token): 
```

Comment:
This logic works only with LambdaDCS. You may want to add rules for the new language too.